Tools/RenovateBot/RenovateHowTo.md

```diff
@@ -0,0 +1,105 @@
+Example:
+```js
+module.exports = {
+    hostRules: [ //Settings on how to connect to your repo
+        {
+            hostType: 'npm',
+            matchHost: 'pkgs.dev.azure.com',
+            username: 'apikey',
+            password: process.env["RENOVATE_TOKEN"],//A devops PAT should go in here.
+        },
+    ],
+    //settings on where to find your repo.
+    endpoint: "https://dev.azure.com/OrganisationName",
+    platform: "azure",
+    repositories: ['OrganisationName/Repo.Name']
+};
+```
+
+## renovate.json
+Stored in the root of your repository, contains settings on how renovate should behave.\
+Example:
+
+```js
+{
+    "extends": [ //Renovate works via a config inheritance scheme with community built recommended defaults.
+        "config:recommended", 
+        "group:recommended" //This is the grouping for which packages share a PR.
+    ],
+    "enabledManagers": ["nuget"], //If you don't specify this it will autodetect and also put in PR's for other package managers.
+    "packageRules": [
+        {
+          "matchUpdateTypes": ["patch", "minor"] //By default renovate just upgrades everything, this is how you make it not do that.
+        }
+    ], 
+    "automerge": false, //By default renovate is happy to merge your PR's for you, if tests pass of course. this is how you make it not.
+    "updateLockFiles": false //I try to use central package management without lockfiles in .net, so there's no lockfiles to update.
+}
+```
+
+# Running it:
+
+Renovate is a Node application, in theory you could just run `npx renovate` once in a while in a configured directory, though you probably want it automated, easiest way is probablyis putting it in a devops yaml pipeline:
+
+```yaml
+schedules:
+- cron: '0 7-20 * * 1-5' #Thanks crontab.guru
+  displayName: Run renovate bot hourly around office hours.
+  branches:
+    include:
+    - main 
+stages:
+- stage: Run
+  jobs:
+    - job: run_renovate
+      displayName: 'Run Renovate'
+      steps:  
+        - task: NodeTool@0
+          inputs:
+             versionSource: 'spec'
+             versionSpec: 20.x
+             checkLatest: true
+        - task: PowerShell@2
+          displayName: 'Run Renovate'         
+          env: #Inject secret as secrets don't go into env by themselves.
+            RENOVATE_TOKEN: $(RENOVATE_TOKEN)
+          inputs:
+            targetType: 'inline' 
+            script: |
+              npm install renovate@38.54.1 #Specific version because this would be pretty bad if it got supply chain attacked.
+              npx renovate
+
+```
+
+# Immortal PR's
+
+Renovate checks what PR's its already created by title, but certain groupings of PR's means it can't be sure it's not already suggested that specific update.\
+When this happens the PR will be marked as immortal. There's settings to prevent this from happening, but you will have to figure them out based on your own config and needs. 
```

Comment:
Could you explain immortal prs?

Reply:
Added some more detail and a quote from the doc's